pkg/rest/apisurface.go

```diff
@@ -387,6 +404,111 @@ func (s *APISurface) UpdateHandler(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, status, response)
 }
 
+func (s *APISurface) GetBindingHandler(
+	w http.ResponseWriter, r *http.Request, asyncBindLogic AsyncBindLogic) {
+	s.Metrics.Actions.WithLabelValues("get_binding").Inc()
+
+	version := getBrokerAPIVersionFromRequest(r)
+	if err := s.BusinessLogic.ValidateBrokerAPIVersion(version); err != nil {
+		writeError(w, err, http.StatusPreconditionFailed)
+		return
+	}
+
+	request, err := unpackGetBindingRequest(r)
+	if err != nil {
+		writeError(w, err, http.StatusBadRequest)
+		return
+	}
+
+	glog.Infof("Received GetBinding request for instanceID %q, bindingID %q", request.InstanceID, request.BindingID)
+
+	c := &broker.RequestContext{
+		Writer:  w,
+		Request: r,
+	}
+
+	response, err := asyncBindLogic.GetBinding(request, c)
+	if err != nil {
+		writeError(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	writeResponse(w, http.StatusOK, response)
+}
+
+func unpackGetBindingRequest(r *http.Request) (*osb.GetBindingRequest, error) {
+	request := &osb.GetBindingRequest{}
+	vars := mux.Vars(r)
+
+	request.InstanceID = vars[osb.VarKeyInstanceID]
+	request.BindingID = vars[osb.VarKeyInstanceID]
+
+	return request, nil
+}
+
+func (s *APISurface) BindingLastOperationHandler(
```

Comment:
I wonder if it would be better for these methods to be attached to the Extender and not the API surface?

Reply:
I went back and forth between keeping all of the related logic together either on the Extender, or at least in their own `extensions.go`. Ultimately I settled on putting the various pieces where their standard counterparts are found. It the various pieces have easy access to the dependencies and helper methods they require without crossing pkg boundaries (I also had a lot of issues with circular deps).

The other option I considered was an extensions package?

Reply:
I like the extension mechanism prosposed here but I don’t think we need it to add async binding support. Can you just add the operations to the main interface? I see the extension thing potentially useful for implementing generic actions - will you make a separate PR for the extensions? I need to think through it a bit more abd don’t want to block async binding support on it.

Reply:
@pmorie ACK, I can implement async bind and then break out the extensions into another proposal PR.